status-im/status-react
# reaction resends under the hood if sending failed

fixes #11020

### Summary

Switching to `status-go` version that silently resends emoji reactions under the hood in case if they weren't successfully sent. 
`status-go` makes up to 3 attempts to resend emoji and waits 30, than 60, than 120 seconds before next resend.

### Testing notes
There is a known issue - if you turned off network connection and immediately sent a reaction - it won't be delivered. The reason is incorrect assigning a "sent" status to messages when network is down but not all peers detected as disconnected yet.
This particular case will be solve in a next `status-go` pr.
Currently should work the case when you turn off the network connection, wait some time (like 10 seconds) and send emoji reaction. When you come online that reaction will be sent again.


#### Platforms
- Android
- iOS
- macOS
- Linux
- Windows

#### Areas that maybe impacted
should be only emoji reactions, but all messages-related functionality better to be tested

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- user profile updates

##### Non-functional
- battery performance
- CPU performance / speed of the app
- network consumption

### Steps to test
- Open Status
- turn off network connection (wait some time)
- add emoji reaction to any message
- at this moment reaction can't be seen on other device
- turn on network connection
- make sure reaction seen on other device after short period of time


status: ready
